### ISSUE TITLE
dart_sass incompatibility with sass with @import and partials (_ prefix)

### ISSUE THREAD

**Issue by bobbens** [nanoc/nanoc#1690]
### Steps to reproduce

1. Set up repository with bootstrap source code.
2. Set up Rule to compile bootstrap source code with dart_sass.
3. Run Nanoc.

### Expected behavior

Clean compilation.

### Actual behavior

Errors on all the `@import` which are trying to include a partial (file starts with `_`).

### Details

The core of the issue is that the old `sass` plugin will cleanly include all these modules that start with `_`, however, the dart_sass one does not, and gives an error on each one. Of course, it should be possible to prepend `_` to all the includes (or add a filter to do that), but for drop-in compatibility, it would make sense to have the same behaviour as sass, especially as this is the latest bootstrap.

```
Sass::CompileError: /home/ess/naev/docs/lore/vendor/bundle/ruby/3.0.0/gems/nanoc-dart-sass-1.0.2/lib/nanoc/dart_sass/filter.rb:74:in `find_item_
for_url': Could not find an item matching pattern `/scss/mixins/banner.*` (RuntimeError)
...

1 │ @import "mixins/banner";
  │         ^^^^^^^^^^^^^^^
  ╵
  nanoc:scss/_bootstrap 1:9  @import
  - 1:9                      root stylesheet
  ...
  ```
  
Note that the file `/scss/mixins/_banner.scss` does exist, but `/scss/mixins/banner.scss` does not.

The same repo cleanly compiles with the `sass` filter. Only does it fail with `dart_sass`.

Reference for partials: https://sass-lang.com/documentation/at-rules/import/#partials

### Crash log

https://gist.github.com/bobbens/017370417a6c1187cb311d26f958702b

**Comment by bobbens** [comment]
The following horrible modification to `vendor/bundle/ruby/3.0.0/gems/nanoc-dart-sass-1.0.2/lib/nanoc/dart_sass/filter.rb` at around line 73  'fixes' it:

```
          unless item
            pat = File.dirname(pat)+"/_"+File.basename(pat)
            item = @items[pat]
            unless item
              raise "Could not find an item matching pattern `#{pat}`"
            end
          end
```

**Comment by denisdefreyne** [comment]
Fixed in #1699. Release coming soon.

**Comment by denisdefreyne** [comment]
This is released in `nanoc-dart-sass` v1.0.3.
